build(deps): summary of dependabot updates

tmberthold opened the pull request.

HeinrichPet commented:
- Implementation ready till 2th December
- Tests and FOSS of CMC ready till 16th December

HeinrichPet commented:
Hi, 

we still need the documentation on how to use RA. Please provide that via /docs/.
And it would be good to merge the base branch, that will fix at least the h2 db failure in the CI Pipeline.

Best regards,
Heinrich

brianjahnke commented:
Hi,
- The Changelog is missing.
- The temporary publishing to local repository needs to be removed.
- The e2e test are failing.
- The changes to the dockerfile seem strange. Would you be so kind to explain why a local copy of the configuration is added as a single layer?

milux commented:
> Hi,
> 
> we still need the documentation on how to use RA. Please provide that via /docs/.
> 
> Best regards, Heinrich

Not mission-critical, since anybody who cannot make sense of the parameters by reading the commends in the `application.properties` file will be able to setup the CMC right now, by any means.
But will provide that as well.

milux commented:
Hi,

> * The Changelog is missing.

I'm not familiar with your expectations regarding your CHANGELOG, especially granularity etc. Could you please elaborate or include a few lines on your own?

> * The temporary publishing to local repository needs to be removed.

Then maven dependency resolution will break, as maven publishing is blocked right now.
The local repository is a sane workaround for that situation, feel free to remove it as soon as maven central is fixed and we were able to publish a fixed release.

> * The e2e test are failing.

I know, and I do neither have any idea why, nor does your pipeline give any useful hint.
Assume broken test or weird side-effect unless you can tell me what's going wrong.

> * The changes to the dockerfile seem strange. Would you be so kind to explain why a local copy of the configuration is added as a single layer?

IDSCP2 library is resolving the certificate paths in a way that prohibits use of JAR-embedded files. These have to be available in the file system, otherwise resolution will fail.

brianjahnke commented:
Hi,
- Here the example [Changelog](https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/main/CHANGELOG.md) for reference.
- This "tmp" fix should not be pushed to main.
- The e2e are reporting a timeout. This means usally the container is not even starting. They can be started locally.
- Why not mount a volume?

milux commented:
Hi,

> * Here the example [Changelog](https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/main/CHANGELOG.md) for reference.

I know the README, but judging from past experiences, I assume that we could have another 3 iterations or so about details in the CHANGELOG if I start working here.
You know best what you expect in your CHANGELOG, in what format, to what granularity. The time of both of us would be wisely invested if you stop urging me updating your CHANGELOG and insert the information yourself. The PR implements RA via the OSS CMC, plus a fix for libs using java.util.logging, so that their logs are getting directed to log4j.

> * This "tmp" fix should not be pushed to main.

I believe that this fix could seriously help any developer who cannot push to official repositories, as it definitely helped me doing so with our libraries.
I just checked, maven central publishing is still not usable, so at the moment you can accept this or wait for Sonatype to fix their infra and then remove this, whatever seems more pleasant to you.

> * The e2e are reporting a timeout. This means usually the container is not even starting. They can be started locally.

If the container was not running locally, I would not have marked this PR for review. So... cannot reproduce, I guess?

> * What prohibits the mounting of a volume?

Nothing, mounting a volume is perfectly fine as well, but additionally increases the effort for any user to try out the feature, moving it even further away from a plug and play solution.
If that's not a problem to you, again, feel free to remove that.

milux commented:
Apparently, just disabling IDSCP2 is sufficient to make `e2e-test` pass.
@ronjaquensel You may want to check this out when you have some minutes for that, I think that merely enabling a feature should not break anything, ideally.

milux commented:
@HeinrichPet Documentation has been provided, see end of `docs/pages/deployment/configuration.md`.

milux commented:
@brianjahnke I accidentally uploaded modifications using experimental buildkit features in the Dockerfile with the last commits. To my great surprise, this did not smash anything in your pipeline.
As it significantly speeds up maven builds locally (as it provides a mounted cache for maven during builds), you may want to consider keeping these changes and instruct users to use `DOCKER_BUILDKIT=1` and `COMPOSE_DOCKER_CLI_BUILD=1` env vars.
Again, if you don't like it, feel free to remove it.

milux commented:
@brianjahnke The local artifacts have now been removed, as the artifacts become available in maven central.
If you think that the local artifact repository doesn't provide any benefit, once again, please feel free to remove that.

I'm done here, I think.

So long, and thanks for all the fish.

dependabot[bot] committed:
chore(deps): Bump pitest-maven from 1.9.9 to 1.9.10 (#538)

Bumps [pitest-maven](https://github.com/hcoles/pitest) from 1.9.9 to 1.9.10.
- [Release notes](https://github.com/hcoles/pitest/releases)
- [Commits](https://github.com/hcoles/pitest/compare/1.9.9...1.9.10)

---
updated-dependencies:
- dependency-name: org.pitest:pitest-maven
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump flyway-core from 9.7.0 to 9.8.1 (#537)

Bumps [flyway-core](https://github.com/flyway/flyway) from 9.7.0 to 9.8.1.
- [Release notes](https://github.com/flyway/flyway/releases)
- [Commits](https://github.com/flyway/flyway/compare/flyway-9.7.0...flyway-9.8.1)

---
updated-dependencies:
- dependency-name: org.flywaydb:flyway-core
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump opentelemetry-semconv (#536)

Bumps [opentelemetry-semconv](https://github.com/open-telemetry/opentelemetry-java) from 1.19.0-alpha to 1.20.0-alpha.
- [Release notes](https://github.com/open-telemetry/opentelemetry-java/releases)
- [Changelog](https://github.com/open-telemetry/opentelemetry-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/open-telemetry/opentelemetry-java/commits)

---
updated-dependencies:
- dependency-name: io.opentelemetry:opentelemetry-semconv
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump camel-idscp2 from 0.11.1 to 0.13.0 (#534)

* chore(deps): Bump camel-idscp2 from 0.11.1 to 0.13.0

Bumps [camel-idscp2](https://github.com/industrial-data-space/idscp2-jvm) from 0.11.1 to 0.13.0.
- [Release notes](https://github.com/industrial-data-space/idscp2-jvm/releases)
- [Commits](https://github.com/industrial-data-space/idscp2-jvm/compare/0.11.1...0.13.0)

---
updated-dependencies:
- dependency-name: de.fhg.aisec.ids:camel-idscp2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* fix: major changes

* fix: major changes in camel routes

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tim Berthold <tim.berthold@sovity.de>

dependabot[bot] committed:
chore(deps): Bump opentelemetry.version from 1.19.0 to 1.20.0 (#535)

Bumps `opentelemetry.version` from 1.19.0 to 1.20.0.

Updates `opentelemetry-api` from 1.19.0 to 1.20.0
- [Release notes](https://github.com/open-telemetry/opentelemetry-java/releases)
- [Changelog](https://github.com/open-telemetry/opentelemetry-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/open-telemetry/opentelemetry-java/compare/v1.19.0...v1.20.0)

Updates `opentelemetry-sdk` from 1.19.0 to 1.20.0
- [Release notes](https://github.com/open-telemetry/opentelemetry-java/releases)
- [Changelog](https://github.com/open-telemetry/opentelemetry-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/open-telemetry/opentelemetry-java/compare/v1.19.0...v1.20.0)

Updates `opentelemetry-exporter-jaeger` from 1.19.0 to 1.20.0
- [Release notes](https://github.com/open-telemetry/opentelemetry-java/releases)
- [Changelog](https://github.com/open-telemetry/opentelemetry-java/blob/main/CHANGELOG.md)
- [Commits](https://github.com/open-telemetry/opentelemetry-java/compare/v1.19.0...v1.20.0)

---
updated-dependencies:
- dependency-name: io.opentelemetry:opentelemetry-api
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: io.opentelemetry:opentelemetry-sdk
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: io.opentelemetry:opentelemetry-exporter-jaeger
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Tim Berthold <75306992+tmberthold@users.noreply.github.com>

Commit:
docs: update changelog
